Refresh.GameServer/Endpoints/ApiV3/LevelApiEndpoints.cs

```diff
@@ -174,4 +175,99 @@ public ApiOkResponse SetLevelAsOverrideByHash(RequestContext context, GameDataba
         
         return new ApiOkResponse();
     }
+
+
+    [ApiV3Endpoint("levels/id/{id}/relations"), MinimumRole(GameUserRole.Restricted)]
+    [DocSummary("Gets your relations to a level by it's ID")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.LevelMissingErrorWhen)]
+    public ApiResponse<ApiGameLevelRelationsResponse> GetLevelRelationsOfUser(RequestContext context, GameDatabaseContext database, GameUser user,
+        [DocSummary("The ID of the level")] int id)
+    {
+        GameLevel? level = database.GetLevelById(id);
+        if (level == null) return ApiNotFoundError.LevelMissingError;
+
+        return new ApiGameLevelRelationsResponse
+        {
+            IsHearted = database.IsLevelFavouritedByUser(level, user),
+            IsQueued = database.IsLevelQueuedByUser(level, user),
+            MyPlaysCount = database.GetTotalPlaysForLevelByUser(level, user)
+        };
+    }
+
+    [ApiV3Endpoint("levels/hearted/username/{username}"), Authentication(false)]
+    [DocSummary("Gets a list of hearted levels by a user by their username")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.UserMissingErrorWhen)]
+    public ApiListResponse<ApiGameLevelResponse> GetLevelsHeartedByUsername(RequestContext context, GameDatabaseContext database, IDataStore dataStore, 
+        [DocSummary("The username of the user")] string username, DataContext dataContext, GameUser? accessingUser) 
+    {
+        GameUser? user = database.GetUserByUsername(username);
+        if(user == null) return ApiNotFoundError.UserMissingError;
+
+        (int skip, int count) = context.GetPageData();
+        DatabaseList<GameLevel> levels = database.GetLevelsFavouritedByUser(user, count, skip, new LevelFilterSettings(context, TokenGame.Website), accessingUser);
+
+        DatabaseList<ApiGameLevelResponse> response = DatabaseList<ApiGameLevelResponse>.FromOldList<ApiGameLevelResponse, GameLevel>(levels, dataContext);
+        return response;
+    } 
```

Comment:
So you want there to rather be an optional "username" query parameter for the "levels/{route}" endpoint for categories like hearted and published levels, which, if set, will get a list of hearted/published/etc levels of that user instead of your own? I don't fully understand what you mean by this.

Reply:
Pretty much, yeah. You can implement it as done in [`ByUserCategory`](https://github.com/LittleBigRefresh/Refresh/blob/main/Refresh.GameServer/Types/Levels/Categories/ByUserLevelCategory.cs)

Reply:
Should be done now.